src/routes/living-form/containers/TestumgebungForm.js

```diff
@@ -0,0 +1,253 @@
+              <MenuItem value={30}>30 Tage</MenuItem>
+            </SelectInput>
+          </StdCol>
+        </Row>
+        <Row>
+          <WideCol>
+            <SingleCheckbox name='agreedToDataProtection'
+                            label={<span>Ich akzeptiere die <a target='_blank'
+                                                               href={'/datenschutz-und-kontakt'}>Datenschutzerklärung</a>.</span>}
+                            validations={[needsAcceptance]} />
+            <LawParagraph>
+              Außerdem willige ich ein, dass der Landkreis Testumgebung und die Tür an Tür - Digital Factory
+              gGmbH meine personenbezogenen Daten zum Zwecke der Wohnraumakquise für anerkannte Flüchtlinge und
+              bleibeberechtigte Migranten erheben, verarbeiten und nutzen. Der Zweck ist ausschließlich
+              auf die Bearbeitung meines Mietangebots beschränkt.
+            </LawParagraph>
+            <LawParagraph>
+              <strong>Hierzu stellt die Tür an Tür - Digital Factory gGmbH meine personenbezogenen Daten auf die
+                öffentliche App Integreat</strong>, die vom Landkreis Testumgebung genutzt wird, um Wohnungen
+              und Mietobjekte zielgerichtet zu vermitteln.
+            </LawParagraph>
+            <LawParagraph>
+              Die Einwilligung kann verweigert bzw. jederzeit ohne Angabe von Gründen über den in der Bestätigungsmail
+              versendeten Link zur dauerhaften Löschung widerrufen werden. In diesem Fall erfolgt keine Bearbeitung
+              meines Mietangebots; alle gespeicherten personenbezogenen Daten werden gelöscht.
+            </LawParagraph>
+          </WideCol>
+        </Row>
+        <Row style={{margin: '20px 0', justifyContent: 'center'}}>
+          <Col center='true'>
+            <FormControl>
+              <SubmitButton validateAll={this.validateAll} type='submit' disabled={this.props.sending}>
+                {this.props.sending ? 'Wird gesendet...' : 'Mietangebot Senden'}
+              </SubmitButton>
+              {this.props.serverError && <FormHelperText error>{this.props.serverError}</FormHelperText>}
+            </FormControl>
+          </Col>
+        </Row>
+      </Grid>
+    </Form>
+  }
+
+  handleSubmit = event => {
+    event.preventDefault()
+    const values = this.state.form.getValues()
+    const requestBody = this.transformFormValuesToRequestBody(values)
+    this.props.sendRequest(requestBody)
+  }
+
+  transformFormValuesToRequestBody (values) {
+    // Transform flat object with properties including a dot to nested object
+    const requestBody = {}
+    forEach(values, (value, key) => set(requestBody, key, value))
+    // Remove customValidation, since it's only a virtual input
+    delete requestBody.customValidation
+    // Make ofAdditionalCosts, ofRooms, ofAdditionalServices arrays (if not already)
+    TestumgebungForm.transformFieldToArray(requestBody.formData.accommodation, 'ofRooms')
```

Comment:
In which cases aren't they arrays?

Reply:
Looks a bit unsafe imo and probably could be made safer with types.

Reply:
For example when only one checkbox is checked. Then the value is only the key of that checkbox... I hate that API... Maybe we'll need/write a wrapper around that...